### PR TITLE
Playwright components testing setup

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
# What

- Playwright ocmponents testing setup
- Added CI jobs for running tests and updating snapshots
- Added examples

# Why

# How Tested

CI, local runs

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/playwright.config.components.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      name: 'Desktop Chrome 16:9',
+      use: {
+        viewport: DESKTOP_16_TO_9_VIEWPORT,
+        launchOptions: { ...chromeLaunchOptions },
+        contextOptions: {
+          userAgent: DESKTOP_USER_AGENT
+        },
+        isStableBuild: notBetaBuildVariable
+      },
+      testMatch: ['OverflowGallery.test.ts']
```
Comment (JamesBurnside) [inline packages/react-components/playwright.config.components.ts:70]: is this necessary?

File: packages/react-components/playwright.config.components.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import {
+  ANDROID_USER_AGENT,
+  DESKTOP_16_TO_9_VIEWPORT,
+  DESKTOP_4_TO_3_VIEWPORT,
+  DESKTOP_USER_AGENT,
+  MINUTE,
+  chromeLaunchOptions
+} from '../../common/config/playwright/playwrightConfigConstants';
+
+const notBetaBuildVariable = process.env['COMMUNICATION_REACT_FLAVOR'] !== 'beta';
```
Comment (JamesBurnside) [inline packages/react-components/playwright.config.components.ts:15]: nit: avoid variable suffix, and negative is perhaps unecessary but thats just opinion

```suggestion
const isBetaBuild = process.env['COMMUNICATION_REACT_FLAVOR'] === 'beta';
```

File: packages/react-components/playwright.config.components.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            babelrc: true,
+            // Use babel.config.js files
+            configFile: true
+          }
+        })
+      ]
+    }
+  },
+
+  projects: [
```
[the remark this answers is not in the document]
Comment (JamesBurnside) [reply packages/react-components/playwright.config.components.ts:48]: Actually I guess not just projects, should be the whole config same as the samplesapp common webpack configs.
```
import { commonPlayrightConfig } from 'somewhereCommon';

export default defineConfig<TestOptions>({
  ...commonPlayrightConfig,
});

// Would be better if playwright supports merged configs something more like:
config = merge(commonPlayrightConfig, { anySpecificConfigSettings });
```

File: common/config/playwright/playwrightConfigConstants.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import path from 'path';
```
Comment (JamesBurnside) [inline common/config/playwright/playwrightConfigConstants.ts:1]: love this file! Great to see all the constants in one easily changeable place!

**Review by JamesBurnside — APPROVED** [review]
This is awesome! Great setup!

File: packages/react-components/tests/browser/TypingIndicator.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    const component = await mount(<TypingIndicator typingUsers={users} />);
+    await component.evaluate(() => document.fonts.ready);
+    await expect(component).toContainText('User 1, User 2, User 3, User 4 and 1 other are typing ...');
+    await expect(component).toHaveScreenshot('typing-indicator-5-users.png');
+  });
+});
+
+// This is an example of beta only test, to be deleted when we have some beta tests examples
+// betaTest.describe('TypingIndicator beta only test', () => {
```
Comment (JamesBurnside) [inline packages/react-components/tests/browser/TypingIndicator.spec.tsx:30]: v cool!

File: packages/react-components/playwright.config.components.ts (hunk 1 above)
Comment (vhuseinova-msft) [reply packages/react-components/playwright.config.components.ts:70]: Good catch! Not needed for components for now

**Review by emlynmac — APPROVED** [review]

**Pull request merged**